### PR TITLE
Added all missing xml docs for Tensors

### PR DESCRIPTION
**Pull request by michaelgsharp**
This PR adds as missing XML docs for all public APIs.

It also enables `UseCompilerGeneratedDocXmlFile` so that we can use these as the source of truth. We will need to review the XML that was already in the repo and not being modified by this PR. That will be done in a separate PR so this is not held by that.

### CODE REVIEW

**Comment by michaelgsharp** [comment]
@gewarren could you review the language for these new docs? We will also need to review the language for the other APIs not being added in this PR, but I'll do another pass on those before I have you take a look.

File: src/libraries/System.Numerics.Tensors/src/System/NIndex.cs [hunk 1]
```diff
@@ -91,7 +91,16 @@ public static NIndex FromEnd(nint value)
             return new NIndex(~value);
         }
 
+        /// <summary>
+        /// Convert the NIndex to an Index.
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/NIndex.cs:95]:
```suggestion
        /// Converts the NIndex to an Index.
```

Comment (gewarren) [reply src/libraries/System.Numerics.Tensors/src/System/NIndex.cs:95]: Ideally these would be crefs too.

File: src/libraries/System.Numerics.Tensors/src/System/NIndex.cs [hunk 2]
```diff
@@ -91,7 +91,16 @@ public static NIndex FromEnd(nint value)
...
+        /// </summary>
+        /// <returns>The converted <see cref="Index"/></returns>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/NIndex.cs:97]:
```suggestion
        /// <returns>The converted <see cref="Index"/>.</returns>
```

Comment (gewarren) [reply src/libraries/System.Numerics.Tensors/src/System/NIndex.cs:97]: In general you don't need to include the type in parameter and return descriptions. It's shown automatically in IntelliSense and docs.

File: src/libraries/System.Numerics.Tensors/src/System/NIndex.cs [hunk 3]
```diff
@@ -91,7 +91,16 @@ public static NIndex FromEnd(nint value)
...
         public Index ToIndex() => checked((Index)this);
+
+        /// <summary>
+        /// Convert the NIndex to an Index without doing bounds checks.
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/NIndex.cs:101]:
```suggestion
        /// Converts the NIndex to an Index without doing bounds checks.
```

File: src/libraries/System.Numerics.Tensors/src/System/NIndex.cs [hunk 4]
```diff
@@ -91,7 +91,16 @@ public static NIndex FromEnd(nint value)
...
+        /// </summary>
+        /// <returns>The converted <see cref="Index"/></returns>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/NIndex.cs:103]:
```suggestion
        /// <returns>The converted <see cref="Index"/>.</returns>
```

File: src/libraries/System.Numerics.Tensors/src/System/NRange.cs [hunk 5]
```diff
@@ -121,12 +121,34 @@ public override string ToString()
...
+        /// <summary>
+        /// Explicitly converts an <see cref="NRange"/> to a <see cref="Range"/>.
+        /// </summary>
+        /// <param name="value"></param>
         public static explicit operator checked Range(NRange value) => new Range(checked((Index)value.Start), checked((Index)value.End));
 
+        /// <summary>
+        /// Converts a <see cref="NRange"/> to a <see cref="Range"/>.
+        /// </summary>
+        /// <returns>The converted <see cref="Range"/></returns>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/NRange.cs:145]:
```suggestion
        /// <returns>The converted <see cref="Range"/>.</returns>
```

File: src/libraries/System.Numerics.Tensors/src/System/NRange.cs [hunk 6]
```diff
@@ -121,12 +121,34 @@ public override string ToString()
...
         public Range ToRange() => new Range(checked((Index)Start), checked((Index)End));
+
+        /// <summary>
+        /// Converts a <see cref="NRange"/> to a <see cref="Range"/> wihout doing bounds checks.
+        /// </summary>
+        /// <returns>The converted <see cref="Range"/></returns>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/NRange.cs:151]:
```suggestion
        /// <returns>The converted <see cref="Range"/>.</returns>
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs [hunk 7]
```diff
@@ -7,40 +7,148 @@
...
     public interface IReadOnlyTensor<TSelf, T> : IEnumerable<T>
         where TSelf : IReadOnlyTensor<TSelf, T>
     {
+        /// <summary>
+        /// Gets an empty tensor.
+        /// </summary>
         static abstract TSelf? Empty { get; }
 
+        /// <summary>
+        /// Gets whether the collection is currently empty.
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:25]:
```suggestion
        /// Gets a value that indicates whether the collection is currently empty.
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs [hunk 8]
```diff
@@ -7,40 +7,148 @@
...
+        /// </summary>
         bool IsEmpty { get; }
+
+        /// <summary>
+        /// Gets whether the underlying buffer pinned.
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:30]:
```suggestion
        /// Gets a value that indicates whether the underlying buffer is pinned.
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs [hunk 9]
```diff
@@ -7,40 +7,148 @@
...
 
+        /// <summary>
+        /// Gets the stride of each dimension in the tensor.
+        /// </summary>
         [UnscopedRef]
         ReadOnlySpan<nint> Strides { get; }
 
+        /// <summary>
+        /// Returns a reference to the 0th element of the tensor. If the tensor is empty, returns null reference.
+        /// It can be used for pinning and is required to support the use of the tensor within a fixed statement.
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:115]:
```suggestion
        /// </summary>
        /// <remarks>
        /// This method can be used for pinning and is required to support the use of the tensor within a fixed statement.
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs [hunk 10]
```diff
@@ -7,40 +7,148 @@
...
+        /// </summary>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:116]:
```suggestion
        /// </remarks>
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs [hunk 11]
```diff
@@ -7,40 +7,148 @@
...
         ReadOnlySpan<nint> Lengths { get; }
 
+        /// <summary>
+        /// Gets the stride of each dimension in the tensor.
+        /// </summary>
         [UnscopedRef]
         ReadOnlySpan<nint> Strides { get; }
 
+        /// <summary>
+        /// Returns a reference to the 0th element of the tensor. If the tensor is empty, returns null reference.
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:114]:
```suggestion
        /// Returns a reference to the 0th element of the tensor. If the tensor is empty, returns <see langword="null" />.
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs [hunk 12]
```diff
@@ -7,40 +7,148 @@
...
+        /// </summary>
+        /// <param name="range">The ranges to be used.</param>
+        /// <returns>The sliced tensor.</returns>
         TSelf Slice(params scoped ReadOnlySpan<NRange> range);
+
+        /// <summary>
+        /// Tries to copy the tensor to the specified destination. The destination tensor must be equal to or larger than the source tensor.
+        /// </summary>
+        /// <param name="destination">The destination span where the data should be copied to.</param>
+        /// <returns>True if the copy succeeded, false otherwise.</returns>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:144]:
```suggestion
        /// <returns><see langword="true" /> if the copy succeeded, <see langword="false" /> otherwise.</returns>
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs [hunk 13]
```diff
@@ -7,40 +7,148 @@
...
         bool TryCopyTo(scoped TensorSpan<T> destination);
+
+        /// <summary>
+        /// Tries to flatten the tensor to the specified destination. The destination span must be equal to or larger than the number of elements in the source tensor.
+        /// </summary>
+        /// <param name="destination">The destination span where the data should be flattened to.</param>
+        /// <returns>True if the copy succeeded, false otherwise.</returns>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:151]:
```suggestion
        /// <returns><see langword="true" /> if the flatten succeeded, <see langword="false" /> otherwise.</returns>
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs [hunk 14]
```diff
@@ -15,25 +20,104 @@ public interface ITensor<TSelf, T>
         // It looks like C#/.NET currently hits limitations here as it believes TSelf and T could be the same type
         // Ideally we could annotate it such that they cannot be the same type and no conflicts would exist
 
+        /// <summary>
+        /// Creates a new tensor with the specified lengths. If <paramref name="pinned"/> is true the underlying buffer is
+        /// created permanently pinned, otherwise the underlying buffer is not pinned. The underlying buffer is initialized to default values.
+        /// </summary>
+        /// <param name="lengths">The lengths of each dimension.</param>
+        /// <param name="pinned">Whether the underlying buffer should be pinned or not. Defaults to False.</param>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:28]:
```suggestion
        /// <param name="pinned"><see langword="true" /> to pin the underlying buffer. The default is <see langword="false" />.</param>
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs [hunk 15]
```diff
@@ -15,25 +20,104 @@ public interface ITensor<TSelf, T>
         // It looks like C#/.NET currently hits limitations here as it believes TSelf and T could be the same type
         // Ideally we could annotate it such that they cannot be the same type and no conflicts would exist
 
+        /// <summary>
+        /// Creates a new tensor with the specified lengths. If <paramref name="pinned"/> is true the underlying buffer is
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:24]: Everything except the first sentence should go in remarks.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs [hunk 16]
```diff
@@ -15,25 +20,104 @@ public interface ITensor<TSelf, T>
...
+        /// created permanently pinned, otherwise the underlying buffer is not pinned. The underlying buffer is initialized to default values.
+        /// </summary>
+        /// <param name="lengths">The lengths of each dimension.</param>
+        /// <param name="pinned">Whether the underlying buffer should be pinned or not. Defaults to False.</param>
         static abstract TSelf Create(scoped ReadOnlySpan<nint> lengths, bool pinned = false);
+
+        /// <summary>
+        /// Creates a new tensor with the specified lengths and strides. If <paramref name="pinned"/> is true the underlying buffer is
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:32]: Everything except the first sentence should go in remarks.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs [hunk 17]
```diff
@@ -15,25 +20,104 @@ public interface ITensor<TSelf, T>
...
+        /// created permanently pinned, otherwise the underlying buffer is not pinned. The underlying buffer is initialized to default values.
+        /// </summary>
+        /// <param name="lengths">The lengths of each dimension.</param>
+        /// <param name="strides">The strides of each dimension.</param>
+        /// <param name="pinned">Whether the underlying buffer should be pinned or not. Defaults to False.</param>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:37]:
```suggestion
        /// <param name="pinned"><see langword="true" /> to pin the underlying buffer. The default is <see langword="false" />.</param>
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs [hunk 18]
```diff
@@ -15,25 +20,104 @@ public interface ITensor<TSelf, T>
...
         static abstract TSelf Create(scoped ReadOnlySpan<nint> lengths, scoped ReadOnlySpan<nint> strides, bool pinned = false);
 
+        /// <summary>
+        /// Creates a new tensor with the specified lengths and strides. If <paramref name="pinned"/> is true the underlying buffer is
+        /// created permanently pinned, otherwise the underlying buffer is not pinned. The underlying buffer is not initialized.
+        /// </summary>
+        /// <param name="lengths">The lengths of each dimension.</param>
+        /// <param name="pinned">Whether the underlying buffer should be pinned or not. Defaults to False.</param>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:45]:
```suggestion
        /// <param name="pinned"><see langword="true" /> to pin the underlying buffer. The default is <see langword="false" />.</param>
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs [hunk 19]
```diff
@@ -15,25 +20,104 @@ public interface ITensor<TSelf, T>
...
+        /// Creates a new tensor with the specified lengths and strides. If <paramref name="pinned"/> is true the underlying buffer is
+        /// created permanently pinned, otherwise the underlying buffer is not pinned. The underlying buffer is initialized to default values.
+        /// </summary>
+        /// <param name="lengths">The lengths of each dimension.</param>
+        /// <param name="strides">The strides of each dimension.</param>
+        /// <param name="pinned">Whether the underlying buffer should be pinned or not. Defaults to False.</param>
         static abstract TSelf Create(scoped ReadOnlySpan<nint> lengths, scoped ReadOnlySpan<nint> strides, bool pinned = false);
 
+        /// <summary>
+        /// Creates a new tensor with the specified lengths and strides. If <paramref name="pinned"/> is true the underlying buffer is
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:41]: Same comment as previous methods.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs [hunk 20]
```diff
@@ -15,25 +20,104 @@ public interface ITensor<TSelf, T>
...
+        /// <param name="pinned">Whether the underlying buffer should be pinned or not. Defaults to False.</param>
         static abstract TSelf CreateUninitialized(scoped ReadOnlySpan<nint> lengths, bool pinned = false);
+
+        /// <summary>
+        /// Creates a new tensor with the specified lengths and strides. If <paramref name="pinned"/> is true the underlying buffer is
+        /// created permanently pinned, otherwise the underlying buffer is not pinned. The underlying buffer is not initialized.
+        /// </summary>
+        /// <param name="lengths">The lengths of each dimension.</param>
+        /// <param name="strides">The strides of each dimension.</param>
+        /// <param name="pinned">Whether the underlying buffer should be pinned or not. Defaults to False.</param>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:54]:
```suggestion
        /// <param name="pinned"><see langword="true" /> to pin the underlying buffer. The default is <see langword="false" />.</param>
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs [hunk 21]
```diff
@@ -15,25 +20,104 @@ public interface ITensor<TSelf, T>
...
+        /// <summary>
+        /// Creates a new tensor with the specified lengths and strides. If <paramref name="pinned"/> is true the underlying buffer is
+        /// created permanently pinned, otherwise the underlying buffer is not pinned. The underlying buffer is not initialized.
+        /// </summary>
+        /// <param name="lengths">The lengths of each dimension.</param>
+        /// <param name="pinned">Whether the underlying buffer should be pinned or not. Defaults to False.</param>
         static abstract TSelf CreateUninitialized(scoped ReadOnlySpan<nint> lengths, bool pinned = false);
+
+        /// <summary>
+        /// Creates a new tensor with the specified lengths and strides. If <paramref name="pinned"/> is true the underlying buffer is
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:49]: Same comment about remarks.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs [hunk 22]
```diff
@@ -15,25 +20,104 @@ public interface ITensor<TSelf, T>
...
+        /// created permanently pinned, otherwise the underlying buffer is not pinned. The underlying buffer is not initialized.
+        /// </summary>
+        /// <param name="lengths">The lengths of each dimension.</param>
+        /// <param name="strides">The strides of each dimension.</param>
+        /// <param name="pinned">Whether the underlying buffer should be pinned or not. Defaults to False.</param>
         static abstract TSelf CreateUninitialized(scoped ReadOnlySpan<nint> lengths, scoped ReadOnlySpan<nint> strides, bool pinned = false);
 
+        /// <summary>
+        /// Gets whether the collection is read-only.
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:58]:
```suggestion
        /// Gets a value that indicates whether the collection is read-only.
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs [hunk 23]
```diff
@@ -15,25 +20,104 @@ public interface ITensor<TSelf, T>
...
         void Clear();
+
+        /// <summary>
+        /// Fills the contents of this tensor with the given value.
+        /// </summary>
         void Fill(T value);
+
+        /// <summary>
+        /// Returns a reference to the 0th element of the tensor. If the tensor is empty, returns null reference.
+        /// It can be used for pinning and is required to support the use of the tensor within a fixed statement.
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:119]:
```suggestion
        /// This method can be used for pinning and is required to support the use of the tensor within a fixed statement.
```

Comment (gewarren) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:119]: Also, this sentence should go in remarks, not summary.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.Factory.cs [hunk 24]
```diff
@@ -12,6 +12,9 @@
 
 namespace System.Numerics.Tensors
 {
+    /// <summary>
+    /// Static class that contains methods for creating tensors.
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.Factory.cs:16]:
```suggestion
    /// Provides methods for creating tensors.
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.Factory.cs [hunk 25]
```diff
@@ -175,6 +178,14 @@ public static Tensor<T> CreateUninitialized<T>(scoped ReadOnlySpan<nint> lengths
             return new Tensor<T>(values, lengths, strides, pinned);
         }
 
+        /// <summary>
+        /// Fills the given <see cref="TensorSpan{T}"/> with random data in a gaussian normal distribution. <see cref="System.Random"/>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.Factory.cs:182]:
```suggestion
        /// Fills the given <see cref="TensorSpan{T}"/> with random data in a Gaussian normal distribution. <see cref="System.Random"/>
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs [hunk 26]
```diff
@@ -614,6 +626,11 @@ public void Dispose()
         }
 
         // REVIEW: PENDING API REVIEW TO DETERMINE IMPLEMENTATION
+        /// <summary>
+        /// Gets the hash code for the <see cref="Tensor{T}"/>.
+        /// </summary>
+        /// <returns>The int hash code of the tensor</returns>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs:632]:
```suggestion
        /// <returns>The hash code of the tensor.</returns>
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs [hunk 27]
```diff
@@ -614,6 +626,11 @@ public void Dispose()
...
+        /// <exception cref="NotImplementedException"></exception>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs:633]:
```suggestion
        /// <exception cref="NotImplementedException">In all cases.</exception>
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs [hunk 28]
```diff
@@ -18,6 +18,9 @@
 
 namespace System.Numerics.Tensors
 {
+    /// <summary>
+    /// Static class that contains methods for tensor operations.
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs:22]:
```suggestion
    /// Provides methods for tensor operations.
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs [hunk 29]
```diff
@@ -6587,10 +6590,17 @@ public static Tensor<T> Xor<T>(in ReadOnlyTensorSpan<T> x, T y)
         }
         #endregion
 
+        /// <summary>
+        /// Gets the smallest broadcastable lengths for two shapes.
+        /// </summary>
+        /// <param name="shape1">The first shape to broadcast.</param>
+        /// <param name="shape2">The second shape to broadcast.</param>
+        /// <returns>The smallest lengths these shapes can be broadcast to.</returns>
+        /// <exception cref="ArgumentException"></exception>
```
Comment (gewarren) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs:6599]:
```suggestion
        /// <exception cref="ArgumentException">The lengths of <paramref name="shape1" /> and  <paramref name="shape2" /> are not broadcast compatible.</exception>
```

**Review by gewarren — COMMENTED** [review]
Thanks for tagging me!

File: src/libraries/System.Numerics.Tensors/src/System.Numerics.Tensors.csproj [hunk 30]
```diff
@@ -6,7 +6,6 @@
     <IsPackable>true</IsPackable>
     <PackageDescription>Provides support for operating over tensors.</PackageDescription>
     <GenAPIExcludeApiList>ReferenceAssemblyExclusions.txt</GenAPIExcludeApiList>
-    <ApiCompatValidateAssemblies>true</ApiCompatValidateAssemblies>
```
Comment (ViktorHofer) [inline src/libraries/System.Numerics.Tensors/src/System.Numerics.Tensors.csproj:9]: @michaelgsharp unrelated change that I smuggled in. This property already defaults to true. No other library sets it explicitly.

**Comment by ViktorHofer** [comment]
@michaelgsharp I resolved your merge conflicts from https://github.com/dotnet/runtime/pull/106179

File: src/libraries/System.Numerics.Tensors/src/System/NIndex.cs (hunk 1 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/NIndex.cs:95]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/NIndex.cs (hunk 3 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/NIndex.cs:101]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/NIndex.cs (hunk 4 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/NIndex.cs:103]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/NRange.cs (hunk 5 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/NRange.cs:145]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/NRange.cs (hunk 6 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/NRange.cs:151]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs (hunk 7 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:25]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs (hunk 8 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:30]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs (hunk 9 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:115]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs (hunk 10 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:116]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs (hunk 11 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:114]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs (hunk 12 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:144]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs (hunk 13 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/IReadOnlyTensor.cs:151]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs (hunk 14 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:28]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs (hunk 15 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:24]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs (hunk 16 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:32]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs (hunk 17 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:37]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs (hunk 19 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:41]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs (hunk 20 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:54]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs (hunk 22 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:58]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs (hunk 23 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:119]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs (hunk 18 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs:45]: Fixed.

File: src/libraries/System.Numerics.Tensors/src/System.Numerics.Tensors.csproj (hunk 30 above)
Comment (michaelgsharp) [reply src/libraries/System.Numerics.Tensors/src/System.Numerics.Tensors.csproj:9]: Yeah, I had it there so I could easily set it to false for offline work since we were adding so many new api's. I'll remove it.

**Comment by michaelgsharp** [comment]
@gewarren all comments have been resolved.

**Review by gewarren — APPROVED** [review]

**Review by carlossanlop — APPROVED** [review]
